### ISSUE TITLE
When a species is compared with itself, why are the gene pairs in the result not all the genes?

### ISSUE THREAD

**Issue by Liuxuanzeng** [tanghaibao/jcvi#713]
Dear Professor,
We want to compare a species with itself to see if there is a gene duplication event, but the results show that there seems to be no gene duplication. We noticed that there are only 6,527 gene pairs in the results, which is not all genes. We would like to ask what parameters can be controlled to display all gene pairs. When comparing a species with itself, all genes should be paired.
![image](https://github.com/user-attachments/assets/fa52b970-bcbd-4c8b-9a98-3ac8fac5b09b)

Best,
Xuanzeng Liu

**Comment by Liuxuanzeng** [comment]
Dear Professor,
The command we run is as follows
python -m jcvi.compara.catalog ortholog Same1 Same2 --no_strip_names

When we analyze two closely related species (Same, Scan), the results will include 13,840 gene pairs.
![image](https://github.com/user-attachments/assets/6050dae5-677f-4ead-bdf7-658b8ed146b2)

Best,
Xuanzeng Liu

**Comment by tanghaibao** [comment]
@Liuxuanzeng 

`jcvi` will remove gene pairs with the same name. 
https://github.com/tanghaibao/jcvi/blob/c9d2a6c58300a0c5b4295c425fce7a90b8768af5/jcvi/compara/blastfilter.py#L57

To run this analysis properly, you'll need to change all the gene names (e.g. add a prefix or suffix) for one of `Same1` and `Same2`.

Please take a look at the `Same1.Same2.anchors` file to see what I meant. The anchors are all different genes.

**Comment by Liuxuanzeng** [comment]
Thanks to the professor, the above problem was solved by renaming all the genes.

We then ran the command python -m jcvi.compara.synteny depth --histogram Same1.Same2.anchors to see the ratio of 1:1 and 1:2 gene pairs. The results are as follows.

Genome Same1 depths:
Depth 0: 19 of 17,236 (0.1%)
Depth 1: 16,953 of 17,236 (98.4%)
Depth 2: 99 of 17,236 (0.6%)
Depth 3: 31 of 17,236 (0.2%)
Depth 4: 76 of 17,236 (0.4%)
Depth 5: 49 of 17,236 (0.3%)
Depth 6: 9 of 17,236 (0.1%)
Genome Same2 depths:
Depth 0: 19 of 17,236 (0.1%)
Depth 1: 16,953 of 17,236 (98.4%)
Depth 2: 99 of 17,236 (0.6%)
Depth 3: 31 of 17,236 (0.2%)
Depth 4: 76 of 17,236 (0.4%)
Depth 5: 49 of 17,236 (0.3%)
Depth 6: 9 of 17,236 (0.1%)
Same1 vs Same2 syntenic depths
1:1 pattern

We want to extract the names of 1:2 or 1:3 gene pairs, how can we obtain them?
There is another small question, what is the difference between the result files Same1.Same2.lifted.anchors and Same1.Same2.anchors? There is not much difference in the graph, but the number of gene pairs is different.


Thanks again!
Xuanzeng Liu
